### ISSUE TITLE
Express plugin not grouping request for wild card routes ( version - 3.3.1 )

### ISSUE THREAD

**Issue by razzpoker50** [DataDog/dd-trace-js#2423]
**Expected behaviour**
Express routes with wild cards should be grouped under a single resource.

**Actual behaviour**
We have an express app had recently upgraded `dd-trace` from `1.5.1` --> `3.3.1`. After upgrading it to the latest version, I started seeing that express routes with wild cards stopped grouping together and shows up as a separate request. 
It used to group them together as expected when I was using `1.5.1`.

Any pointers as to how best to fix the issue would be great. 

**Environment** 
* **Node.js version:** 16
* **Tracer version:** 3.3.1

**End points**
```
   app.get(`${this.basePath}/saved-homes/`, ....);
   app.post(`${this.basePath}/saved-home/`, ....)
   app.get(`${this.basePath}/saved-home-ids/`, ....)
   app.delete(`${this.basePath}/saved-homes/:homeId`, ....)
```

**redacted package.json**
```
"dependencies": {
    "dd-trace": "3.3.1",
    "opentracing": "^0.14.4"
  },
```

**Setting Resource Name**
```
    const getResourceName = req => `${req.method} ${req.url.split('?')[0]}`,

    // Override the resource name.
    span.addTags({ 'resource.name': getResourceName(req) });
```

**After Upgrading to latest** 
<img width="1021" alt="Resource Name" src="https://user-images.githubusercontent.com/2760513/193683452-33648351-df76-4f15-93a3-f60437e5e770.png">

**Before the Upgrade**
<img width="849" alt="Resource Name Before" src="https://user-images.githubusercontent.com/2760513/193683775-be6d0c9d-b1f5-4f3e-a6fa-726e1baea4c8.png">

**Comment by rochdev** [comment]
It looks like you're overriding the resource name, what happens if you remove the override?

**Comment by razzpoker50** [comment]
I am in the process of checking that and start relying on the `req.route` property to see if that takes care of it. 
Curious as to why it use to work with version `1.5.1` even though we were overwriting the  resourceName.

**Comment by rochdev** [comment]
It's possible that there was a bug in dd-trace and the default ended up overriding the override since we usually recommend changing the `http.route` tag and not the resource name directly to do this. However, in your case it sounds like you're actually looking for what we do by default, so the override should not be needed in the first place.

**Comment by razzpoker50** [comment]
Thanks for the explanation. Will keep posted on the findings once I test it out after explicitly setting the `resource.name` tag.

**Comment by razzpoker50** [comment]
@rochdev I ended up updating the method to use `route` object instead to create the resource names. The reason I did not get rid of explicitly setting `resource.name` is because there are other applications using the the same library and the ask was to provide an option to override `resource.name` with a custom function.

```
const getResourceName = req => `${req.method} ${req.route.path}`,
```

Fell free to close the bug report and thank you for looking into it. 
<img width="855" alt="Screen Shot 2022-10-04 at 12 29 19 PM" src="https://user-images.githubusercontent.com/2760513/193908695-eb7c11a0-bbbd-412c-9518-ad8778423706.png">
